src/main/java/app/servlets/AddRecordServlet.java

```diff
@@ -0,0 +1,190 @@
+                List items = null;
+                try {
+                    items = upload.parseRequest(new ServletRequestContext(req));
+                }
+                catch (FileUploadException e) {
+                    e.printStackTrace();
+                }
+                Iterator itr = items.iterator();
+                while (itr.hasNext())
+                {
+                    FileItem item = (FileItem) itr.next();
+                    if (item.isFormField())
+                    {
+                        String name = item.getFieldName();
+                        String value = item.getString();
+                        if(name.equals("title"))
+                        {
+                            title=value;
+                        }
+                        if(name.equals("author"))
+                        {
+                            author=value;
+                        }
+                        if(name.equals("genre"))
+                        {
+                            genre=value;
+                        }
+                        if(name.equals("publisher"))
+                        {
+                            publisher=value;
+                        }
+                        if(name.equals("date"))
+                        {
+                            date=value;
+                        }
+                        if(name.equals("description"))
+                        {
+                            description=value;
+                        }
+                        if(name.equals("pages"))
+                        {
+                            pages=Integer.parseInt(value);
+                        }
+                        if(name.equals("isbn"))
+                        {
+                            isbn=Long.parseLong(value);
+                        }
+                        if(name.equals("total_amount"))
+                        {
+                            total_amount=Integer.parseInt(value);
+                        }
+                    }
+                    else
+                    {
+                        int id = 1;
+                        filename = new File(item.getName()).getName();
+                        int i = filename.lastIndexOf('.');
+                        if (i != -1) {
+                            String filename2 = filename.substring(0, i) + id;
+                            id++;
+                            String ext = filename.substring(i, filename.length());
+                            filename = filename2 + ext;
+                        }
+                        try {
+                            file = new File(UPLOAD_DIRECTORY + File.separator + filename);
+                            item.write(file);
+                            byte[] bArray = readFileToByteArray(file);
+                            String base64Image = Base64.encodeBase64String(bArray);
+                            book.setCover64(base64Image);
```

Comment:
you store book cover as a row in a table and as an actual file? You should pick one way